<launch>
  <arg name="rsb_transport_socket_enabled" default="1" />
  <arg name="rsb_transport_spread_enabled" default="0" />
  <arg name="rsb_transport_socket_server" default="auto" />
  <arg name="rsb_transport_socket_host" default="localhost" />
  <arg name="rsb_transport_socket_port" default="4444" />
  <arg name="table_safety_margin" default="0.0" />
  <arg name="camera_frame" default="" />
  <arg name="cloud_frame" default="base_link" />
  <arg name="obj_fitter_configs" default="$(find sq_fitting_adapter)/config/default_config.yaml" />
  <arg name="rednum" default="100" /> <!-- sq fitter reduces points in cloud to this number (if not zero) -->
  <rosparam command="load" file="$(arg obj_fitter_configs)" />

  <!-- object fitter -->
  <node pkg="sq_fitting_adapter" type="vision_adapter_node.py" name="object_fitter" args="" output="screen" >
    <param name="table_safety_margin" type="double" value="$(arg table_safety_margin)"/>
    <param name="camera_frame" type="string" value="$(arg camera_frame)"/>
    <param name="cloud_frame" type="string" value="$(arg cloud_frame)"/>
    
  </node>

  <node pkg="sq_fitting_ros" type="fitting_service.py" name="sq_fitting" args="-r $(arg rednum)" output="screen" >
  </node>

</launch>
